### PR TITLE
remove hardcoded URL, add instructions, fix tests

### PR DESCRIPTION
**Pull request by martindstone**
## Background

Confluence plugin had a hardcoded URL, marketplace plugins need to work without being recompiled.

## This PR

Removed the hardcoded URL and added code to get the URL from a plugin config entity; added instructions to create the plugin config entity if it's not found.

### Security

- [X] Security impact of change has been considered
- [X] Code follows company security practices and guidelines

### CODE REVIEW

**Review by cristinabuenahora — APPROVED** [review]

File: plugins/confluence-plugin/src/components/App.test.tsx [hunk 1]
```diff
@@ -1,53 +1,41 @@
-import { render, screen, waitFor } from "@testing-library/react";
+// import { render, screen, waitFor } from "@testing-library/react";
```
[the remark this answers is not in the document]
Comment (martindstone) [reply plugins/confluence-plugin/src/components/App.test.tsx]: ✅

File: plugins/confluence-plugin/src/components/PageContent.tsx [hunk 2]
```diff
@@ -12,16 +10,45 @@ import {
...
+  const [baseConfluenceUrl, setBaseConfluenceUrl] = useState<string>("");
+  const [errorStr, setErrorStr] = useState<string>("");
 
   const context = usePluginContext();
 
   useEffect(() => {
+    if (!context?.apiBaseUrl) {
+      return;
+    }
+    const getConfluencePluginConfig = async (): Promise<void> => {
```
Comment (jreock) [inline plugins/confluence-plugin/src/components/PageContent.tsx:28]: I think in the long run we'll want to add an abstraction into @cortexapps/plugin-core to retrieve configuration bundles, but for our PoC I think this works, just adding the comment for posterity.

Comment (martindstone) [reply plugins/confluence-plugin/src/components/PageContent.tsx:28]: agreed, ideally we would have some boilerplate to show if various things went horribly wrong

File: plugins/confluence-plugin/README.md [hunk 3]
```diff
@@ -6,25 +6,42 @@ This plugin makes it possible to view Confluence assets alongside related entiti
...
+### Find your Confluence instance URL
 
-```
-const baseConfluenceUrl = "https://confluence-server.atlassian.net";
-```
+Your Confluence instance URL should look like `https://something.atlassian.net`. We will use this URL in the plugin's proxy and in the plugin's configuration entity.
+
+### Set your Confluence credentials
+
+If you are using username and password authentication, type them in to a text editor with a colon (`:`) between them. If you are using SSO, use an API token in place of the password. To create an API token in Confluence by clicking on Security > Create and manage API tokens > Create API token. Once you have your text in your text editor like `myusername@example.com:MySecretPassWordOrToken`, you need to base64 encode it. You can do this using an online tool like [this](https://www.base64encode.org). You will add the base64 encoded value to Cortex as a secret. Copy the base64 encoded value. In Cortex, click on Settings > Secrets > Add secret. In Secret name, type `confluence_secret`, paste the base64 encoded secret into Secret value, and click on Create secret.
```
Comment (jreock) [inline plugins/confluence-plugin/README.md:17]: Nitpick, typo, should be: "To create an API tolein in Confluence, click on Security > Create"

Comment (jreock) [inline plugins/confluence-plugin/README.md:17]: Also, depending on the version and tenant type of Confluence, they may not have an ability to create the API Keys from Confluence Settings and instead will need to navigate to Profile Menu (upper right) -> Account -> Security -> API Tokens. We could provide both, and then link to Atlassian Docs: https://support.atlassian.com/atlassian-account/docs/manage-api-tokens-for-your-atlassian-account/

**Review by jreock — APPROVED** [review]
LGTM!

**Pull request merged**